LanguageFeatures/Patterns/execution_pattern_await_for_in_statement_A02_t03.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// A statement of the form:
+///
+/// for (<keyword> <pattern> in <expression>) <statement>
+///
+/// Where <keyword> is var or final is treated like so:
+/// 1. Let I be the static type of <expression>, inferred using context type
+///   schema Iterable<P> where P is the context type schema of <pattern>.
+/// 2. Calculate the element type of I:
+///   i. If I implements Iterable<T> for some T then E is T.
+///   ii. Else if I is dynamic then E is dynamic.
+///   iii. Else it is a compile-time error.
+/// 3. Type check <pattern> with matched value type E.
+/// 4. If there are no compile-time errors, then execution proceeds as the
+///   following code, where id1 and id2 are fresh identifiers:
+/// ```dart
+/// var id1 = <expression>;
+/// var id2 = id1.iterator;
+/// while (id2.moveNext()) {
+///   <keyword> <pattern> = id2.current;
+///   { <statement> }
+/// }
+/// ```
+/// @description Checks that in an async for-in loop if `I` is `dynamic` and
+/// runtime type of `I` is `Stream<T>` where `T` is not assignable to
+/// `<pattern>` required type then a run-time error occurs
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns,records
+
+import "../../Utils/expect.dart";
+import "patterns_lib.dart";
+
+main() async {
+  Expect.throws(() {
+    await for (var (int v1)
+        in Stream<num>.fromIterable([1, 2, 3.14]) as dynamic) {}
```

Comment:
Can we verify that the run-time error occurs because the stream is a `Stream<num>` where `num` is not assignable to `int`, and it does _not_ occur when we get the first actual element type mismatch?

For example, the body of the loop could add the elements to a log, and then we'd expect an empty log rather than a log that contains 1 and 2 (and then there was an exception with 3.14).

Reply:
Got a log now, thanks!